src/libcore/ops.rs

```diff
@@ -1486,7 +1486,7 @@ shr_assign_impl_all! { u8 u16 u32 u64 usize i8 i16 i32 i64 isize }
 #[lang = "index"]
 #[rustc_on_unimplemented = "the type `{Self}` cannot be indexed by `{Idx}`"]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub trait Index<Idx: ?Sized> {
+pub trait Index<Idx: ?Sized = usize> {
```

Comment:
Type inference is not good enough to infer
```
let a: u16 = 10;
let b = vector[a.into()];
```

But... I've just tested it and it turns out the inference is not good enough to infer it even with `trait Index<Idx: ?Sized = usize>` and `#![feature(default_type_parameter_fallback)]`! So, this change can be reverted for now.